CardanoSharp.Wallet.Test/TransactionTests.cs

```diff
@@ -689,6 +687,63 @@ public void CertificateTest()
                 serialized.ToStringHex());
         }
 
+        [Theory]
+        [InlineData("2eb6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", true)]
+        [InlineData("e02eb6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", false)]
+        [InlineData("b6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", false)]
+        public void CertificateBuilder_SetStakeRegistration_Tests(string stakeAddressHex, bool expectedSuccess)
+        {
+            var bytes = stakeAddressHex.HexToByteArray();
+            var actualSuccess = false;
+            try
+            {
+                CertificateBuilder.Create.SetStakeRegistration(bytes);
+                actualSuccess = true;
+            }
+            catch
+            { }
+            Assert.Equal(expectedSuccess, actualSuccess);
+        }
+
+        [Theory]
+        [InlineData("2eb6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", true)]
+        [InlineData("e02eb6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", false)]
+        [InlineData("b6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", false)]
+        public void CertificateBuilder_SetStakeDeregistration_Tests(string stakeAddressHex, bool expectedSuccess)
+        {
+            var bytes = stakeAddressHex.HexToByteArray();
+            var actualSuccess = false;
+            try
+            {
+                CertificateBuilder.Create.SetStakeDeregistration(bytes);
+                actualSuccess = true;
+            }
+            catch
+            { }
+            Assert.Equal(expectedSuccess, actualSuccess);
+        }
+
+        [Theory]
+        [InlineData("2eb6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", "ad4396a0f7c47e3d69ee8bd792c80ca0bffee61945cec5c42b4ae90f", true)]
+        [InlineData("e02eb6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", "ad4396a0f7c47e3d69ee8bd792c80ca0bffee61945cec5c42b4ae90f", false)]
+        [InlineData("2eb6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", "e0ad4396a0f7c47e3d69ee8bd792c80ca0bffee61945cec5c42b4ae90f", false)]
+        [InlineData("b6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", "ad4396a0f7c47e3d69ee8bd792c80ca0bffee61945cec5c42b4ae90f", false)]
+        [InlineData("2eb6d138fda3f907baba4d3e10ddc0cf18e4287c848759841909cf01", "4396a0f7c47e3d69ee8bd792c80ca0bffee61945cec5c42b4ae90f", false)]
+        public void CertificateBuilder_SetStakeDelegation_Tests(string stakeAddressHex, string stakePoolIdHex, bool expectedSuccess)
+        {
+            var stakeBytes = stakeAddressHex.HexToByteArray();
+            var poolBytes = stakePoolIdHex.HexToByteArray();
+            var actualSuccess = false;
+            try
+            {
+                CertificateBuilder.Create.SetStakeDelegation(stakeBytes, poolBytes);
+                actualSuccess = true;
+            }
+            catch
+            { }
+            Assert.Equal(expectedSuccess, actualSuccess);
```

Comment:
Just a minor thing on all of this if you want to trim up code, you can assert that a method was not thrown like so https://github.com/xunit/xunit/issues/188#issuecomment-104378832. Completely optional just wanted to make you aware :)